File: Algorithms/General_Algorithms/TripletSum Algorithm/tripletSum.c
#include <stdio.h> 

bool tripletSum(int array[], int arr_size, int sum) 
{ 
	int l, r; 

	for (int i = 0; i < arr_size - 2; i++) { 

		for (int j = i + 1; j < arr_size - 1; j++) { 

			for (int k = j + 1; k < arr_size; k++) { 
				if (array[i] + array[j] + array[k] == sum) { 
					printf("Triplet is %d, %d, %d", 
						array[i], array[j], array[k]); 
					return true; 
				} 
			} 
		} 
	} 

	return false; 
} 

int main() 
{ 
	int array[100];
	int sum,i,arr_size;
  	printf("Size of array:");
  	n=scanf("%d",&arr_size);
  	printf("Sum for triplet you are looking for:");
  	n=scanf("%d",&sum);
  	
  	for(i=0;i<n;i++){
    	printf("enter %dth value",i);
      	sacnf("%d",&array[i]);
    }
 
	tripletSum(array, arr_size, sum); 
	return 0; 
} 

File: Algorithms/General_Algorithms/Triplet_Sum/TripletSum.py
'''
if there is a triplet in array whose sum is equal to the given value print the triplet and return true
the sum is stored in sum variable
''' 
 

def triplet_sum(A, arr_size, sum): 

	# fix one number as i 
	for i in range( 0, arr_size-2): 

		# Fix second number as j
		for j in range(i + 1, arr_size-1): 
			
			# Looking for the third number
			for k in range(j + 1, arr_size): 
				if A[i] + A[j] + A[k] == sum: 
					print("Triplet is", A[i], 
						", ", A[j], ", ", A[k]) 
					return True
	
	# If we reach here, then there is no triplet in the list
	return False
A = [1, 4, 45, 6, 10, 8] 
sum = 22
arr_size = len(A) 
triplet_sum(A, arr_size, sum) 

